src/internet.ts

```diff
@@ -34,42 +35,60 @@ export class Internet {
    * @param firstName The optional first name to use. If not specified, a random one will be chosen.
    * @param lastName The optional last name to use. If not specified, a random one will be chosen.
    * @param provider The mail provider domain to use. If not specified, a random free mail provider will be chosen.
+   * @param config The optional config to use. If not specified, configuration options will not be applied.
    *
    * @example
    * faker.internet.email() // 'Kassandra4@hotmail.com'
    * faker.internet.email('Jeanne', 'Doe') // 'Jeanne63@yahoo.com'
    * faker.internet.email('Jeanne', 'Doe', 'example.fakerjs.dev') // 'Jeanne_Doe88@example.fakerjs.dev'
+   * faker.internet.email('Jeanne', 'Doe', 'example.fakerjs.dev', { allowSpecialCharacters: true }) // 'Jeanne%Doe88@example.fakerjs.dev'
    */
-  email(firstName?: string, lastName?: string, provider?: string): string {
+  email(
+    firstName?: string,
+    lastName?: string,
+    provider?: string,
+    config?: EmailConfig
+  ): string {
     provider =
       provider ||
       this.faker.random.arrayElement(
         this.faker.definitions.internet.free_email
       );
-    return (
-      this.faker.helpers.slugify(
-        this.faker.internet.userName(firstName, lastName)
-      ) +
-      '@' +
-      provider
+    let localPart: string = this.faker.helpers.slugify(
+      this.faker.internet.userName(firstName, lastName)
     );
+    if (config && config.allowSpecialCharacters) {
+      const usernameChars: string[] = '._-'.split('');
+      const specialChars: string[] = ".!#$%&'*+-/=?^_`{|}~".split('');
+      localPart = localPart.replace(
+        this.faker.random.arrayElement(usernameChars),
+        this.faker.random.arrayElement(specialChars)
+      );
+    }
+    return localPart + '@' + provider;
```

Comment:
We have another PR were we start to try to always use template literals
```suggestion
    return `${localPart}@${provider}`;
```